[org-toggle-latex-overlays> ](org-toggle-latex-overlays> )[org-toggle-pretty-entities](org-toggle-pretty-entities)




# Reaction rates and rate laws




## The rate of a reaction



-   We are now in a position to define the rate of a reaction as:

$R^* = \frac{dX}{dt}$ [eq-rate](eq-rate)

-   which is the rate of the change of the extensive reaction extent. Note that this is:

-   Independent of any particular species
-   Dependent on how the reaction is written

-   The second point is a result of how we defined $X$.

-   For the reaction $2 \ce{H_2} + \ce{O_2} \rightarrow 2 \ce{H_{2}O}$ we defined:

$\frac{N_{A_1} - N_{A_1,0}}{-2} = \frac{N_{A_2} - N_{A_2,0}}{-1} = \frac{N_{A_3} - N_{A_3,0}}{2} = X $

-   For the reaction $\ce{H_2} + 1/2 \ce{O_2} \rightarrow \ce{H_{2}O}$ we can define:

$\frac{N_{A_1} - N_{A_1,0}}{-1} = \frac{N_{A_2} - N_{A_2,0}}{-0.5} = \frac{N_{A_3} - N_{A_3,0}}{1} = X' $

-   You can see that $X$ depends on the stoichiometric coefficients, so we have to know the reaction and how it was written when we discuss reaction rates.

-   $X$ is an extensive variable with units of moles

Exercise: Suppose we start with 2 moles of oxygen at t=0 in a reactor. Later,  there is 1.4 moles of oxygen remaining. Calculate the extent of reaction for these two reactions. Use the extent to determine how much hydrogen is remaining.

1.  $2 \ce{H_2} + \ce{O_2} \rightarrow 2 \ce{H_{2}O}$

2.  $\ce{H_2} + 1/2 \ce{O_2} \rightarrow \ce{H_{2}O}$




## The rate of disappearance of a reactant



-   Now, recalling that $X = \frac{N_{A_j} - N_{A_j,0}}{\alpha_j}$, we arrive at:

$ \frac{dX}{dt} = \frac{1}{\alpha_j} \frac{dN_{A_J}}{dt} $

or rearranging, and using [eq-rate](eq-rate):

$ \frac{dN_{A_J}}{dt} = \alpha_j R^* $

-   Thus, the change in composition of species J *due to the reaction* is just the stoichiometric coefficient for that species times the reaction rate. If the volume is constant:

$ r = R^* / V = \frac{d\xi}{dt} $

This defines the intensive reaction rate, with units of mol / vol / time.

That also leads to (only for constant volume!)

$ \frac{dC_{A_J}}{dt} = \alpha_j r$

-   We define the species rate of production as:

$r_j = \alpha_j r$

-   This is **NOT** the rate law!
    -   It is a relationship between the rate of appearance of a species and the reaction rate
    
    -   One of our goals is to find the function that describes the rate of the reaction and its dependence on concentration and temperature
    
    -   This will come later in the course.




## Rates of disappearance and appearance of other species



-   The stoichiometric coefficients define the rates of appearance and disappearance of other species.

$ r = \frac {r_{A_1}}{\alpha_1} = \frac{r_{A_2}}{\alpha_2}  = \frac{r_{A_3}}{\alpha_3} \cdots $

-   Remember that stoichiometric coefficients of reactants are negative, and products are positive

-   We call $r$ the rate of the reaction with units (typically) mol / vol / time

-   The rate of disappearance of species $A_1$ is $r_{A_1} = \alpha_1 r$

-   The rate generally depends on the concentration of reactants (and sometimes products), as well as the temperature of the reactor.




### Practice



The ethylene epoxidation reaction has the following stoichiometry:

C2H4 + 1/2 O2  -> C2H4O

If we have a reactor with one mole of ethylene and one mole of oxygen, which reactant is stoichiometrically limiting?

[ethylene]((message-box "Correct"))     [oxygen]((message-box "Incorrect. How many moles of ethylene are needed to consume one mole of oxygen?"))

If oxygen is consumed at 1 mol/min, what is the rate of ethylene consumption?

[1/2 mol/min]((message-box "Incorrect"))  [1 mol/min]((message-box "Not correct"))  [2 mol/min]((message-box "Correct"))

What is the maximum amount of ethylene oxide that can be produced?

[1/2 mol]((message-box "Not correct"))  [1 mol]((message-box "Correct."))  [2 mol]((message-box "Not correct"))




## Rate laws



-   The rate law is an algebraic equation that relates the rate of reaction to the concentrations of reactants, products and temperature.

-   Law of mass action for elementary steps:
    -   Reaction rate is proportional to the concentration of each reactant raised to its stoichiometric coefficient
    -   That is the definition of an elementary reaction
    -   For example: $A + B \rightarrow C$
        -   $r = k C_A C_B$
    
    -   $2A \rightarrow B$
        -   $r = k C_A^2$

-   Many other more complex rate laws exist for non-elementary reactions
    -   $r = \frac {k_1 C_A}{1 + k_2 C_A}$ e.g. for surface reactions or enzyme reactions
    
    -   $r = k C_A^{3/2}$ for complex mechanisms

-   Rate laws are ultimately determined from experiments

-   We use these rate laws in conjunction with stoichiometry and mole balances to design reactors.




# Introduction to ODEs in python



We solve ordinary differential equations in the form  of $f'(X) = f(X, f, f')$. We will use numerical integration to solve these kinds of equations. These basically work by integrating the equations from some known initial condition on some range of the independent variable.  [odeint](odeint)

[scipy.integrate.odeint](https://docs.scipy.org/doc/scipy/reference/generated/scipy.integrate.odeint.html)

Let us consider $\frac{dy}{dx} = y(x)$ with $y(0) = 1$. Let us evaluate the solution from x=0 to x=2. The solution to this ODE is simply $y(x) = e^x$, and we plot that for comparison.

We have to define a function in Python to define f(X, f, f').



In [1]:
import numpy as np
from scipy.integrate import odeint
import matplotlib.pyplot as plt

def dydx(y, x):
    return y

xspan = np.linspace(0, 2)
y0 = 1

sol = odeint(dydx, y0, xspan)

plt.plot(xspan, sol)
plt.plot(xspan, np.exp(xspan), 'r--')
plt.xlabel('x')
plt.ylabel('y')
plt.legend(['numerical solution', 'analytical solution'], loc='best')
plt.savefig('soln.png')

![img](./soln.png)

Note the use of a dashed line on the second plot so that it is apparent the red dashed line lies on top of the blue line, and the use of the legend to show which line is which.

Suppose we want to find the value of x that makes the solution equal to 6? You can estimate that by reading off of the graph above, but we can do better by using an event to stop the integration. We use a different integration function for this. We have to define a new event function that is equal to zero at the value of y we want.

[pycse.odelay](https://www.google.com/#safe=off&q=pycse.odelay)            [pycse.odelay](pycse.odelay)



In [1]:
import numpy as np
from pycse import odelay
import matplotlib.pyplot as plt

def dydx(y, x):
    '$\frac{dy}{dx} = y$'
    return y

def event(y, x):
    'event for y=6'
    value = y - 6  # zero at the y-value we want
    isterminal = True
    direction = 0       # all zeros
    return value, isterminal, direction

xspan = np.linspace(0, 2)
y0 = 1

x, y, te, ye, ie =  odelay(dydx, y0, xspan, [event])
print('At x = {0:1.2f} y = {1}'.format(te[0], ye[0]))

At x = 1.79 y = [ 6.]

This tells us an event occurred at x ≈ 1.79, and at that x, y=6. We can check that analytically in this case:



In [1]:
import numpy as np
print(np.log(6))

1.79175946923

Remember, [numpy.log](https://docs.scipy.org/doc/numpy/reference/generated/numpy.log.html) is the natural log!

Next time we will review the mole balances. I strongly recommend you review [mole-balance.md>](mole-balance.md>)before coming to class. We will cover a lot of material.

